plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -277,6 +299,33 @@ fieldSuggestCodeAction recorder ide _ (CodeActionParams _ _ (TextDocumentIdentif
       let completionTexts = fmap (^. JL.label) completions
       pure $ FieldSuggest.fieldErrorAction uri fieldName completionTexts _range
 
+cabalAddCodeAction :: Recorder (WithPriority Log) -> PluginMethodHandler IdeState 'LSP.Method_TextDocumentCodeAction
+cabalAddCodeAction recorder state plId (CodeActionParams _ _ (TextDocumentIdentifier uri) _ CodeActionContext{_diagnostics=diags}) = do
```

Comment:
I think this code action requires a little bit of code re-organisation.

The issue is that some expensive operations are executed, even though we don't know whether we even need them. 

The current structure is:
1. Find a .cabal file for a given filepath
2. Parse said cabal file
3. You try to guess the component
4. Only now you check whether the diagnostic is even a hidden package description!

This means, for each button click where the client asks for code actions, we perform the steps (1)-(3). Please change this such that (4) happens first :)

Reply:
Oh, good catch! I have missed that, thank you.